flink-streaming-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/proxy/KinesisProxy.java

```diff
@@ -155,55 +199,79 @@ public GetRecordsResult getRecords(String shardIterator, int maxRecordsToGet) {
 	 * @param startingSeqNum the sequence number that the iterator will start from
 	 * @return the shard iterator
 	 */
+	@Override
 	public String getShardIterator(KinesisStreamShard shard, String shardIteratorType, String startingSeqNum) {
-		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
+		return kinesisClient.getShardIterator(shard.getStreamName(), shard.getShard().getShardId(), shardIteratorType, startingSeqNum).getShardIterator();
```

Comment:
@rmetzger This method doesn't do the `ProvisionedThroughputExceededException` check / retries like the `getRecords()` method. Perhaps we should be checking that + fail hard after retrying too, as it also has a 5 requests per second limit? Reasoning is the same as my reply to your comment on the `getRecords()` method.